content/sensu-core/0.29/guides/troubleshooting-sensu.md

```diff
@@ -0,0 +1,336 @@
+
+**Sensu Logs**:
+{{< highlight json >}}{"timestamp":"2018-06-10T16:39:15.988000+0200","level":"warn","message":"transport connection error","reason":"tcp connection lost"}
+{"timestamp":"2018-06-10T16:39:15.989000+0200","level":"warn","message":"transport connection error","reason":"possible authentication failure. wrong credentials?","user":"sensu"}{{< /highlight >}}
+
+Much like the errors seen in the previous section, the failure to connect to RabbitMQ appears to be one related to credentials. However, we can go a bit deeper by looking at the RabbitMQ logs, which present an error similar to the following:
+
+{{< highlight shell >}}2018-06-11 15:31:03.515 [info] <0.1540.0> TLS server: In state certify at ssl_handshake.erl:1289 generated SERVER ALERT: Fatal - Handshake Failure - {bad_cert,invalid_ext_key_usage}{{< /highlight >}}
+
+_NOTE: We'll presume that if you've gone through our SSL guide, that you're using the [SSL tool][8] to generate the certificates used in your deployment. If not, this is not a problem, as the commands we'll use for troubleshooting this particular scenario will prove useful no matter how your cert and key pairs are generated._
+
+Let's start off by manually verifying our certificate and key pairs. Sensu's SSL tool will place the certs/keys in the following directory:
+
+{{< highlight shell >}}
+sensu_ssl_tool
+├── client
+│ ├── cert.pem
+│ ├── keycert.p12
+│ ├── key.pem
+│ └── req.pem
+├── sensu_ca
+│ ├── cacert.cer
+│ ├── cacert.pem
+│ ├── certs
+│ │ ├── 01.pem
+│ │ └── 02.pem
+│ ├── index.txt
+│ ├── index.txt.attr
+│ ├── index.txt.attr.old
+│ ├── index.txt.old
+│ ├── openssl.cnf
+│ ├── private
+│ │ └── cakey.pem
+│ ├── serial
+│ └── serial.old
+├── server
+│ ├── cert.pem
+│ ├── keycert.p12
+│ ├── key.pem
+│ └── req.pem
+└── ssl_certs.sh{{< /highlight >}}
+
+So from that directory, we'll ensure that our server cert and key (read "cert and key that are being used inside of the RabbitMQ configuration") match
+
+{{< highlight shell >}}
+$ openssl x509 -noout -modulus -in server/cert.pem | openssl md5
+(stdin)= 32df80471e2d4e7d0453f60cfb66b2b2
+$ openssl rsa -noout -modulus -in server/key.pem | openssl md5
+(stdin)= 32df80471e2d4e7d0453f60cfb66b2b2{{< /highlight >}}
+
+And then the same with our client cert and key (cert and key being used inside of the Sensu configuration):
+
+{{< highlight shell >}}
+$ openssl x509 -noout -modulus -in client/cert.pem | openssl md5
+(stdin)= c2a6a5a28a629653741e7674c3b95b19
+$ openssl rsa -noout -modulus -in client/key.pem | openssl md5
+(stdin)= c2a6a5a28a629653741e7674c3b95b19{{< /highlight >}}
+
+_WARNING: Should the values here NOT match, you'll need to regenerate your cert/key pairs._
+
+And just to test to ensure that we can indeed connect with our cert/key pairs, we can use openssl to connect to our RabbitMQ instance directly:
+
+{{< highlight shell >}}
+$ openssl s_client -connect localhost:5671 -key client/key.pem{{< /highlight >}}
```

Comment:
Curious about your views on whether to prepend commands with `$`

Reply:
Do we have a sense for what convention we're using most consistently? I suspect that we aren't actually using `$` or `#` many example commands.

Reply:
More often, we aren't using them. Per our slack convo, we'll be leaving out these symbols since users may not know their meanings and they complicate the copy-paste process.